### PR TITLE
MerkleDB Compact Path Bytes Nits

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

See comments

## How this works

Self-explanatory

## How this was tested

Existing UT

### CODE REVIEW

File: x/merkledb/codec.go [hunk 1]
```diff
@@ -327,43 +327,43 @@ func (*codecImpl) decodeID(src *bytes.Reader) (ids.ID, error) {
...
-	_, _ = dst.Write(s.Bytes())
+func (c *codecImpl) encodePath(dst *bytes.Buffer, p Path) {
+	c.encodeUint(dst, uint64(p.length))
+	_, _ = dst.Write(p.Bytes())
 }
 
 func (c *codecImpl) decodePath(src *bytes.Reader, branchFactor BranchFactor) (Path, error) {
 	if minPathLen > src.Len() {
-		return EmptyPath(branchFactor), io.ErrUnexpectedEOF
+		return Path{}, io.ErrUnexpectedEOF
```
Comment (danlaine) [inline x/merkledb/codec.go:337]: No need for a function call when we're returning a non-nil error anyway

File: x/merkledb/db.go [hunk 2]
```diff
@@ -217,9 +216,10 @@ func newDatabase(
 		rootGenConcurrency = config.RootGenConcurrency
 	}
 
-	if config.BranchFactor == BranchFactorUnspecified {
-		return nil, ErrNoSpecifiedBranchFactor
+	if err := config.BranchFactor.Valid(); err != nil {
```
Comment (danlaine) [inline x/merkledb/db.go:219]: Might as well check for _all_ invalid branch factors instead of just one

File: x/merkledb/path.go [hunk 3]
```diff
@@ -16,32 +18,10 @@ const (
 	Byte   byte = 8
 )
 
-type BranchFactor int
```
Comment (danlaine) [inline x/merkledb/path.go:19]: Changed so that package-level var declarations come first

File: x/merkledb/path.go [hunk 4]
```diff
@@ -16,32 +18,10 @@ const (
...
-
-const (
-	BranchFactorUnspecified BranchFactor = 0
```
Comment (danlaine) [inline x/merkledb/path.go:22]: Removed this since I don't think we need this specific invalid branch factor anymore

File: x/merkledb/path.go [hunk 5]
```diff
@@ -67,12 +47,41 @@ var (
...
 )
 
+func (f BranchFactor) Valid() error {
+	if _, ok := branchFactorToPathConfig[f]; ok {
+		return nil
+	}
+	return fmt.Errorf("%w: %d", errInvalidBranchFactor, f)
+}
+
+func EmptyPath(bf BranchFactor) Path {
```
Comment (danlaine) [inline x/merkledb/path.go:68]: No reason for this to not be a normal function (rather than a var) afaict

**Review by darioush — APPROVED** [review]

**Pull request merged**